syncd/VirtualOidTranslator.cpp

```diff
@@ -83,6 +83,16 @@ sai_object_id_t VirtualOidTranslator::translateRidToVid(
         SWSS_LOG_THROW("RID 0x%" PRIx64 " is switch object, but not in local or redis db, bug!", rid);
     }
 
+    if((object_type == SAI_OBJECT_TYPE_PORT) && (switchVid == SAI_NULL_OBJECT_ID))
```

Comment:
please remove this from here, notice that you are introducing bug here, since this translation function will refer to all ports if switch is null

Reply:
Shouldnt allocateNewObjectId() be skipped from notification?

Reply:
no, since when syncd is doing nothing it can still generate some notification with internal object, we try to make sure that will not happen when discovering new objects

Reply:
maybe this is a good point, to not allocate new object, i would need to think about this

Reply:
please remove this

Reply:
please remove this check, you dont need it any more